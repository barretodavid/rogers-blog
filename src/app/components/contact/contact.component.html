<form [formGroup]="form" (ngSubmit)="onSubmit()">
  <mat-card>
    <mat-card-title>Contact Form</mat-card-title>
    <mat-card-content class="flex flex-column">
      <app-input type="text" label="Name" formControlName="name"></app-input>
      <app-input type="email" label="Email" formControlName="email"></app-input>
      <app-textarea label="Message" formControlName="message"></app-textarea>
    </mat-card-content>
    <mat-card-actions>
      <button mat-raised-button color="primary" type="submit" [disabled]="form.invalid">Save</button>
      <button mat-raised-button type="button" (click)="onClear()">Clear</button>
    </mat-card-actions>
  </mat-card>
</form>