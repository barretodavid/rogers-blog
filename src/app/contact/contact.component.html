<form [formGroup]="form" (ngSubmit)="onSubmit()">
  <mat-card>
    <mat-card-title>Contact Form</mat-card-title>
    <mat-card-content>

      <app-input type="text" label="Name" formControlName="name"></app-input>
      <app-input type="email" label="Email" formControlName="email"></app-input>

      <!-- <mat-form-field>
        <mat-label>Email</mat-label>
        <input type="email" matInput formControlName="email">
        <mat-error *ngIf="email.hasError('required')">This field is required</mat-error>
        <mat-error *ngIf="email.hasError('email')">This field should be an email</mat-error>
      </mat-form-field> -->

      <mat-form-field>
        <mat-label>Message</mat-label>
        <textarea matInput formControlName="message"></textarea>
        <mat-error *ngIf="message.hasError('required')">This field is required</mat-error>
        <mat-error *ngIf="message.hasError('minlength')">This field needs to have at least 10 characters</mat-error>
      </mat-form-field>

    </mat-card-content>
    <mat-card-actions>
      <button mat-raised-button color="primary" type="submit" [disabled]="form.invalid">Save</button>
      <button mat-raised-button type="button" (click)="onClear()">Clear</button>
    </mat-card-actions>
  </mat-card>
</form>