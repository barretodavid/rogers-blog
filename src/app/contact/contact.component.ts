import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, AbstractControl, Validators } from '@angular/forms';

@Component({
  selector: 'app-contact',
  templateUrl: './contact.component.html',
  styleUrls: ['./contact.component.scss']
})
export class ContactComponent implements OnInit {
  form: FormGroup;

  constructor(private fb: FormBuilder) {}

  ngOnInit() {
    this.form = this.fb.group({
      name: ['', Validators.required],
      email: ['', [Validators.required, Validators.email]],
      message: ['', [Validators.required, Validators.minLength(10)]]
    });
  }

  get name(): AbstractControl {
    return this.form.get('name');
  }

  get email(): AbstractControl {
    return this.form.get('email');
  }

  get message(): AbstractControl {
    return this.form.get('message');
  }

  onSubmit(): void {
    console.log(this.form.value);
  }

  onClear(): void {
    this.form.reset();
    // this.name.hasError('email')
  }

}
