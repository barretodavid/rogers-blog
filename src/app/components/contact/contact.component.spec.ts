import { ContactComponent } from './contact.component';
import { TestBed } from '@angular/core/testing';
import { ReactiveFormsModule, NgControl, FormControl } from '@angular/forms';
import { MatCardModule, MatButtonModule, MatFormFieldModule, MatInputModule } from '@angular/material';
import { InputComponent } from '../input.component';
import { TextareaComponent } from '../textarea.component';
import { Component } from '@angular/core';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

const nameControl = new FormControl('David');
const messageControl = new FormControl('message');

describe('ContactComponent', () => {
  it('should render its template', () => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        MatCardModule,
        MatButtonModule,
        MatFormFieldModule,
        MatInputModule,
        NoopAnimationsModule,
      ],
      declarations: [
        ContactComponent,
        InputComponent,
        TextareaComponent,
      ]
    })
    .overrideComponent(InputComponent, {
      set: {
        providers: [{
          provide: NgControl,
          useValue: nameControl,
        }]
      }
    })
    .overrideComponent(TextareaComponent, {
      set: {
        providers: [{
          provide: NgControl,
          useValue: messageControl,
        }]
      }
    })
    .compileComponents().then(() => {
      const fixture = TestBed.createComponent(ContactComponent);
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });
  });
});

