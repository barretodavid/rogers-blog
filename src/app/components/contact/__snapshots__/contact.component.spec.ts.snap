// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactComponent should render its template 1`] = `
<app-contact
  fb={[Function FormBuilder]}
  form={[Function FormGroup]}
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <mat-card
      class="mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Contact Form
      </mat-card-title>
      <mat-card-content
        class="flex flex-column mat-card-content"
      >
        <app-input
          class="ng-untouched ng-pristine ng-invalid"
          formcontrolname="name"
          label="Name"
          ng-reflect-label="Name"
          ng-reflect-name="name"
          ng-reflect-type="text"
          type="text"
        >
          <mat-form-field
            class="w-100 mat-form-field ng-tns-c2-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
          >
            <div
              class="mat-form-field-wrapper"
            >
              <div
                class="mat-form-field-flex"
              >
                
                
                <div
                  class="mat-form-field-infix"
                >
                  <input
                    aria-invalid="false"
                    aria-required="false"
                    class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                    id="mat-input-0"
                    matinput=""
                    ng-reflect-form="[object Object]"
                    ng-reflect-type="text"
                    type="text"
                  />
                  <span
                    class="mat-form-field-label-wrapper"
                  >
                    
                    <label
                      aria-owns="mat-input-0"
                      class="mat-form-field-label ng-tns-c2-0 mat-empty mat-form-field-empty ng-star-inserted"
                      for="mat-input-0"
                      id="mat-form-field-label-1"
                      ng-reflect-disabled="true"
                      ng-reflect-ng-switch="true"
                    >
                      
                      
                      <mat-label
                        class="ng-star-inserted"
                      >
                        Name
                      </mat-label>
                      
                      
                    </label>
                  </span>
                </div>
                
              </div>
              
              <div
                class="mat-form-field-underline ng-tns-c2-0 ng-star-inserted"
              >
                <span
                  class="mat-form-field-ripple"
                />
              </div>
              <div
                class="mat-form-field-subscript-wrapper"
                ng-reflect-ng-switch="hint"
              >
                
                
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c2-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  
                  <div
                    class="mat-form-field-hint-spacer"
                  />
                </div>
              </div>
            </div>
          </mat-form-field>
        </app-input>
        <app-input
          class="ng-untouched ng-pristine ng-invalid"
          formcontrolname="email"
          label="Email"
          ng-reflect-label="Email"
          ng-reflect-name="email"
          ng-reflect-type="email"
          type="email"
        >
          <mat-form-field
            class="w-100 mat-form-field ng-tns-c2-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
          >
            <div
              class="mat-form-field-wrapper"
            >
              <div
                class="mat-form-field-flex"
              >
                
                
                <div
                  class="mat-form-field-infix"
                >
                  <input
                    aria-invalid="false"
                    aria-required="false"
                    class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                    id="mat-input-1"
                    matinput=""
                    ng-reflect-form="[object Object]"
                    ng-reflect-type="email"
                    type="email"
                  />
                  <span
                    class="mat-form-field-label-wrapper"
                  >
                    
                    <label
                      aria-owns="mat-input-1"
                      class="mat-form-field-label ng-tns-c2-1 mat-empty mat-form-field-empty ng-star-inserted"
                      for="mat-input-1"
                      id="mat-form-field-label-3"
                      ng-reflect-disabled="true"
                      ng-reflect-ng-switch="true"
                    >
                      
                      
                      <mat-label
                        class="ng-star-inserted"
                      >
                        Email
                      </mat-label>
                      
                      
                    </label>
                  </span>
                </div>
                
              </div>
              
              <div
                class="mat-form-field-underline ng-tns-c2-1 ng-star-inserted"
              >
                <span
                  class="mat-form-field-ripple"
                />
              </div>
              <div
                class="mat-form-field-subscript-wrapper"
                ng-reflect-ng-switch="hint"
              >
                
                
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c2-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  
                  <div
                    class="mat-form-field-hint-spacer"
                  />
                </div>
              </div>
            </div>
          </mat-form-field>
        </app-input>
        <app-textarea
          class="ng-untouched ng-pristine ng-invalid"
          formcontrolname="message"
          label="Message"
          ng-reflect-label="Message"
          ng-reflect-name="message"
        >
          <mat-form-field
            class="w-100 mat-form-field ng-tns-c2-2 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
          >
            <div
              class="mat-form-field-wrapper"
            >
              <div
                class="mat-form-field-flex"
              >
                
                
                <div
                  class="mat-form-field-infix"
                >
                  <textarea
                    aria-invalid="false"
                    aria-required="false"
                    class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                    id="mat-input-2"
                    matinput=""
                    ng-reflect-form="[object Object]"
                  />
                  <span
                    class="mat-form-field-label-wrapper"
                  >
                    
                    <label
                      aria-owns="mat-input-2"
                      class="mat-form-field-label ng-tns-c2-2 mat-empty mat-form-field-empty ng-star-inserted"
                      for="mat-input-2"
                      id="mat-form-field-label-5"
                      ng-reflect-disabled="true"
                      ng-reflect-ng-switch="true"
                    >
                      
                      
                      <mat-label
                        class="ng-star-inserted"
                      >
                        Message
                      </mat-label>
                      
                      
                    </label>
                  </span>
                </div>
                
              </div>
              
              <div
                class="mat-form-field-underline ng-tns-c2-2 ng-star-inserted"
              >
                <span
                  class="mat-form-field-ripple"
                />
              </div>
              <div
                class="mat-form-field-subscript-wrapper"
                ng-reflect-ng-switch="hint"
              >
                
                
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c2-2 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  
                  <div
                    class="mat-form-field-hint-spacer"
                  />
                </div>
              </div>
            </div>
          </mat-form-field>
        </app-textarea>
      </mat-card-content>
      <mat-card-actions
        class="mat-card-actions"
      >
        <button
          class="mat-raised-button mat-primary _mat-animation-noopable"
          color="primary"
          disabled=""
          mat-raised-button=""
          ng-reflect-color="primary"
          ng-reflect-disabled="true"
          type="submit"
        >
          <span
            class="mat-button-wrapper"
          >
            Save
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="true"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        <button
          class="mat-raised-button _mat-animation-noopable"
          mat-raised-button=""
          type="button"
        >
          <span
            class="mat-button-wrapper"
          >
            Clear
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
      </mat-card-actions>
    </mat-card>
  </form>
</app-contact>
`;
